app/controllers/course/group/group_categories_controller.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+class Course::Group::GroupCategoriesController < Course::ComponentController
+  include Course::Group::GroupManagerConcern
+
+  load_resource :group_category, class: Course::GroupCategory
```

Comment:
Why not use load_and_authorize_resource here?

Reply:
Hmm I think it's because I wanted to be specific in what I wanted to authorize it for, for each specific route. Any advice on this?

Reply:
For show_info and show_users - I think you can instead add those into the ability components. 

Sth like: can [ :read, :show_info, :show_users ], Course::GroupCategory, course_id: course.id

Since the rest are either read or manage, it will be automatically specified (index and show use read, whereas the rest use manage).

Reply:
Thanks for the suggestion! Changed.